app/code/community/Drip/Connect/Helper/Customer.php

```diff
@@ -22,18 +24,33 @@ public function proceedAccount(
             $this->getLogger()->log("Skipping guest subscriber update due to unusable email", Zend_Log::NOTICE);
             return;
         }
-
         $customerData = Drip_Connect_Helper_Data::prepareCustomerData($customer, true, $forceStatus, $acceptsMarketing);
+        $subscriberRequest = new Drip_Connect_Model_ApiCalls_Helper_CreateUpdateSubscriber($customerData, $config);
+        $subscriberRequest->call();
+        $apiCall = new Drip_Connect_Model_ApiCalls_Helper_RecordAnEvent($config, array(
+            'email' => $customer->getEmail(),
+            'action' => $event,
+        ));
+        $response = $apiCall->call();
+    }
 
-        Mage::getModel('drip_connect/ApiCalls_Helper_CreateUpdateSubscriber', $customerData)->call();
+    /**
+     * Gets the first store when a customer is in website scope.
+     * @param Mage_Customer_Model_Customer $customer
+     * @return string Customer ID
+     */
+    public function firstStoreIdForCustomer(Mage_Customer_Model_Customer $customer) {
```

Comment:
nit: firstStoreIdForCustomer seems to named according the the fallback implementation.  It seems better to go with the more generic naming - something like getCustomerStoreId?

Reply:
That sounds great to me!